apps/chat/src/utils/app/application.ts

```diff
@@ -190,3 +194,38 @@ export const createQuickAppConfig = ({
     QUICK_APP_CONFIG_DIVIDER,
   );
 };
+
+export const topicToOption = (topic: string) => ({
+  value: topic,
+  label: topic,
+  ...getTopicColors(topic),
+});
+
+export const isExecutableApp = (entity: DialAIEntityModel) =>
+  !!entity.functionStatus;
+
+export const getApplicationType = (entity: DialAIEntityModel) => {
+  if (isQuickApp(entity)) return ApplicationType.QUICK_APP;
+  if (isExecutableApp(entity)) return ApplicationType.CODE_APP;
+
+  return ApplicationType.CUSTOM_APP;
+};
+
+export const getApplicationNextStatus = (entity: DialAIEntityModel) => {
+  return entity.functionStatus === ApplicationStatus.STARTED
+    ? ApplicationStatus.STOPPING
+    : ApplicationStatus.STARTING;
+};
+
+export const getApplicationSimpleStatus = (entity: DialAIEntityModel) => {
+  switch (entity.functionStatus) {
+    case ApplicationStatus.CREATED:
+    case ApplicationStatus.STOPPED:
+    case ApplicationStatus.FAILED:
+      return 'start';
```

Comment:
enum ApplicationSimpleStatus?

Reply:
May be an extra enum, it's more like util for shortening other implementations like icons and captions.

Reply:
You don't use it as labels or something it's some special values which are you using as mapping to icons and for classes.
It's much better to use enum-values instead in all places than hard-coded strings